lib/dispatcher.lib.php

```diff
@@ -356,6 +356,65 @@ public function updateuserstats($dispatcher, &$reqData, &$out) {
 		$dispatcher->setPrefix(''); // No need for prefix since only usable by server.
 	}
 
+	public function updatenamecolor($dispatcher, &$reqData, &$out) {
```

Comment:
where is the permissions check for this?

Reply:
The permissions check on this end validates that it comes from sim3. Sim3 itself can handle permissions checking, since only that server (including an IP check) can send said request.

Reply:
Yeah, it validates both the server token and the IP address the request is coming from.